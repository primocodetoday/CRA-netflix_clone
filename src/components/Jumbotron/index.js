import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
