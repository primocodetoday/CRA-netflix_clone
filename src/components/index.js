export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
