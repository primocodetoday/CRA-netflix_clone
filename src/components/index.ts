export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
