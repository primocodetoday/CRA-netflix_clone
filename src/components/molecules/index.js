export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
