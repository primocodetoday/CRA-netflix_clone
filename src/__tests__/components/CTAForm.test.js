import React from 'react';
import { render, screen } from '@testing-library/react';

import { CTAForm } from 'components';

describe('<CTAForm/> renders properly', () => {
  it('with data', () => {
    const { container } = render(
      <CTAForm>
        <CTAForm.Text>TestString</CTAForm.Text>
        <CTAForm.Input placeholder="Email address" />
        <CTAForm.Button>TRY 30 DAYS FREE</CTAForm.Button>
      </CTAForm>,
    );

    expect(screen.getByText('TestString')).toBeTruthy();
    expect(screen.getByText('TRY 30 DAYS FREE')).toBeTruthy();
    expect(screen.getByPlaceholderText('Email address')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
