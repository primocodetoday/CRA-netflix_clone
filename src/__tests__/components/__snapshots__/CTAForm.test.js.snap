// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CTAForm/> renders properly with data 1`] = `
<div
  class="StyledCTAForm__Container-lhj0cj-0 cBMAWP"
>
  <h3
    class="StyledCTAForm__Content-lhj0cj-2 jwsvuj"
  >
    TestString
  </h3>
  <input
    class="StyledCTAForm__Input-lhj0cj-1 encAPy"
    placeholder="Email address"
  />
  <button
    class="StyledCTAForm__Button-lhj0cj-3 eLPgHr"
  >
    TRY 30 DAYS FREE
    <img
      alt="Get Started"
      src="/assets/icons/chevron-right.png"
    />
  </button>
</div>
`;
